grammars/javascript.cson

```diff
@@ -1023,6 +1004,29 @@
     ]
   }
   {
+    'begin': '\\?'
+    'beginCaptures':
+      '0':
+        'name': 'keyword.operator.ternary.js'
+    'end': ':'
+    'endCaptures':
+      '0':
+        'name': 'keyword.operator.ternary.js'
+    'patterns': [
+      {
+        'match': '(\\w+)(?=\\s*:)'
```

Comment:
Can you explain what this match does in more detail?

In addition, why is the second `$self` needed?